pkg/clusterprovider/providers/kstone/kstone.go

```diff
@@ -217,22 +217,57 @@ func (c *EtcdClusterKstone) Equal(cluster *kstonev1alpha1.EtcdCluster) (bool, er
 		return false, nil
 	}
 
-	oldCPU, _, _ := unstructured.NestedString(etcd.Object, "spec", "template", "resources", "requests", "cpu")
-	if oldCPU != strconv.Itoa(int(cluster.Spec.TotalCpu)) {
-		klog.Info("cpu is different")
+	oldCPULimits, _, _ := unstructured.NestedString(etcd.Object, "spec", "template", "resources", "limits", "cpu")
+	if oldCPULimits == "" {
+		oldCPULimits = "0"
+	}
+	desiredCPULimits := cluster.Spec.Resources.Limits.Cpu().String()
+	if oldCPULimits != desiredCPULimits {
+		klog.Info("cpu limits is different")
 		return false, nil
 	}
 
-	oldMemory, _, _ := unstructured.NestedString(
+	oldMemoryLimits, _, _ := unstructured.NestedString(
+		etcd.Object,
+		"spec",
+		"template",
+		"resources",
+		"limits",
+		"memory",
+	)
+	if oldMemoryLimits == "" {
+		oldMemoryLimits = "0"
+	}
+	desiredMemoryLimits := cluster.Spec.Resources.Limits.Memory().String()
```

Comment:
cluster.Spec.Resources.Limits.Memory() may be nil pointer.

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/resource.go#L57